tests/Mremi/UrlShortener/Tests/Provider/Sina/SinaProviderTest.php

```diff
@@ -0,0 +1,478 @@
+    {
+        $this->mockClient($this->getMockResponseWithNoId(), 'get');
+
+        $this->provider->expand($this->getBaseMockLink());
+    }
+
+    /**
+     * Tests the expand method throws exception if Sina returns a response with no longUrl.
+     *
+     * @expectedException        \Mremi\UrlShortener\Exception\InvalidApiResponseException
+     * @expectedExceptionMessage Property "longUrl" does not exist within Sina response.
+     */
+    public function testExpandThrowsExceptionIfApiResponseHasNoLongUrl()
+    {
+        $this->mockClient($this->getMockResponseWithNoLongUrl(), 'get');
+
+        $this->provider->expand($this->getBaseMockLink());
+    }
+
+    /**
+     * Tests the expand method throws exception if Sina returns an invalid status code.
+     *
+     * @expectedException        \Mremi\UrlShortener\Exception\InvalidApiResponseException
+     * @expectedExceptionMessage Sina returned status code "21506" with message "Error: Parameter value is not valid!
+     */
+    public function testExpandThrowsExceptionIfApiResponseHasInvalidStatusCode()
+    {
+        $response = $this->getBaseMockResponse();
+
+        $apiRawResponse = <<<'JSON'
+{
+"error": "Error: Parameter value is not valid!",
+"error_code": 21506,
+"request": "/2/sinaurl/public/expand.json"
+}
+JSON;
+
+        $stream = $this->getBaseMockStream();
+        $stream
+            ->expects($this->once())
+            ->method('getContents')
+            ->will($this->returnValue($apiRawResponse));
+
+        $response
+            ->expects($this->once())
+            ->method('getBody')
+            ->will($this->returnValue($stream));
+
+        $this->mockClient($response, 'get');
+
+        $this->provider->expand($this->getBaseMockLink());
+    }
+
+    /**
+     * Tests the expand method with a valid Sina's response.
+     */
+    public function testExpandWithValidApiResponse()
+    {
+        $response = $this->getBaseMockResponse();
+
+        $apiRawResponse = <<<'JSON'
+{
+    "urls": [
+        {
+            "result": true,
+            "url_short": "http://t.cn/h5ef4",
+            "url_long": "http://www.google.com/",
+            "transcode": 0,
+            "type": 39
+        }
+    ]
+}
+JSON;
+
+        $stream = $this->getBaseMockStream();
+        $stream
+            ->expects($this->once())
+            ->method('getContents')
+            ->will($this->returnValue($apiRawResponse));
+
+        $response
+            ->expects($this->once())
+            ->method('getBody')
+            ->will($this->returnValue($stream));
+
+        $link = $this->getMockShortLink();
+        $link
+            ->expects($this->once())
+            ->method('setLongUrl')
+            ->with($this->equalTo('http://www.google.com/'));
+
+        $this->mockClient($response, 'get');
+
+        $this->provider->expand($link);
+    }
+
+    /**
+     * Initializes the provider.
+     */
+    protected function setUp()
```

Comment:
Can you move it at the top of the class please (same for `tearDown`)? By convention, these methods are declared first to increase readability.